pyccel/codegen/wrapper/c_to_python_wrapper.py

```diff
@@ -782,7 +792,7 @@ def _wrap_FunctionDef(self, expr):
         # is an ndarray.
         for a in original_c_args:
             orig_var = getattr(a, 'original_function_argument_variable', a.var)
-            v = self.scope.find(self.scope.get_expected_name(orig_var.name), category='variables')
+            v = self.scope.find(orig_var.name, category='variables')
```

Comment:
why we are not using `self.scope.get_expected_name` anymore

Reply:
The `find` function of `Scope` is used to get a `Variable` object from the `PyccelSymbol` containing its name. The `PyccelSymbol` contains the name used in Python, while the `Variable` contains the name used in the target language. The 2 objects are mapped to one another with `Scope._locals` which is populated in `insert_variable`. The second argument of `insert_variable` ensures that the correct symbol is used for the key. As mentioned in the PR this argument was missing in several places so this look-up wasn't working correctly. I seem to have got muddled up when originally writing this as instead of adding the missing arguments I added calls to `get_expected_name`.

The function `get_expected_name` is used to get the name of an object in the target language from its Python name. This is usually used in the semantic stage to create a `Variable` it should not be used to call `find`.

Does this make sense? I guess the current docs for the Scope in [developer_docs](https://github.com/pyccel/pyccel/blob/devel/developer_docs/scope.md) and the docstrings are not sufficient?

Reply:
now i got it i haven't interacted with the `find` much
in the times i did i always used `get_expected_name`
it is more clear now.

Reply:
I think the docs are very clear now. Can we resolve this conversation, @bauom?